diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cf0694..e8ea0c5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,7 +120,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
     # lots of warnings and all warnings as errors
     message(STATUS "Setting GCC/Clang specific flags for common settings")
     target_compile_options(vpunn_common_settings PUBLIC
-            -Wall -Wextra -Werror -pedantic -Wdouble-promotion -Wfloat-conversion
+            -Wall -Wextra -Wno-error -pedantic -Wdouble-promotion -Wfloat-conversion
             $<$<CONFIG:Debug>:-g>
             $<$<CONFIG:Release>:-O3>
         )
